### DISCUSSION TITLE
how to add  background video

### DISCUSSION THREAD

**Discussion by Menna-elgallad** [fabricjs/fabric.js#8809] [General]
what is the proper way to make background video and fit it in fabric canvas while scale if not fit but without stretching

**Comment by iocoker** [comment]
First, find the smaller of ratio of the `canvas width / video width` vs `canvas height / video height`
Then you can [scaleToWidth](http://fabricjs.com/docs/fabric.Image.html#scaleToWidth) or [scaleToHeight](http://fabricjs.com/docs/fabric.Image.html#scaleToHeight) the video to whichever is lower. Width or Height ratio
